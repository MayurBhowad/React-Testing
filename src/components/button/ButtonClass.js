import React, { Component } from 'react'

export class ButtonClass extends Component {

    constructor(props) {
        super(props);
        this.state = {
            myClass: 'mouseOut'
        }

        this._onMouseEnter = this._onMouseEnter.bind(this)
        this._onMouseLeave = this._onMouseLeave.bind(this)
    }

    _onMouseEnter() {
        this.setState({ myClass: 'hovered' })
    }

    _onMouseLeave() {
        this.setState({ myClass: 'mouseOut' })
    }

    render() {
        return (

            <button className={`one, two, ${this.state.myClass}`}
                onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}
            >{this.props.children}</button>

        )
    }
}

export default ButtonClass
