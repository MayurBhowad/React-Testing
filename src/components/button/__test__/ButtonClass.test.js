import React from 'react';
import renderer from 'react-test-renderer';
import ButtonClass from '../ButtonClass';


test('Button changes the class on hover', () => {
    const component = renderer.create(
        <ButtonClass>this is Button</ButtonClass>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter();
    tree = component.toJSON()
    expect(tree).toMatchSnapshot();

    tree.props.onMouseLeave();
    tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
