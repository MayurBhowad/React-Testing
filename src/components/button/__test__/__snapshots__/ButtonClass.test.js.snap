// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes the class on hover 1`] = `
<button
  className="one, two, mouseOut"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  this is Button
</button>
`;

exports[`Button changes the class on hover 2`] = `
<button
  className="one, two, hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  this is Button
</button>
`;

exports[`Button changes the class on hover 3`] = `
<button
  className="one, two, mouseOut"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  this is Button
</button>
`;
